group-coordinator/src/test/java/org/apache/kafka/coordinator/group/runtime/InMemoryPartitionWriter.java

```diff
@@ -102,6 +103,14 @@ public long append(
         }
     }
 
+    @Override
+    public long completeTransaction(
+        TopicPartition tp,
+        WriteTxnMarkersRequest.TxnMarkerEntry marker
+    ) throws KafkaException {
+        throw new IllegalStateException("Not implemented");
```

Comment:
What is this class used for again? And is this planned in a future PR?

Reply:
It is only used in tests. I will implement it if needed while adding tests.